aodntools/timeseries_products/hourly_timeseries.py

```diff
@@ -0,0 +1,581 @@
+                data_codes.append(get_data_code(parameter))
+
+            nc_clean = in_water(nc)  # in water only
+            nc_clean = good_data_only(nc_clean, qcflags)  # good quality data only
+            df_metadata = df_metadata.append({'source_file': file,
+                                              'instrument_id': nc.attrs['deployment_code'] + '; ' + nc.attrs[
+                                                  'instrument'] + '; ' + nc.attrs['instrument_serial_number'],
+                                              'LONGITUDE': nc.LONGITUDE.squeeze().values,
+                                              'LATITUDE': nc.LATITUDE.squeeze().values,
+                                              'NOMINAL_DEPTH': get_nominal_depth(nc)},
+                                             ignore_index=True)
+
+            df_temp = nc_clean.to_dataframe()
+
+            ## keep TIME as the only index
+            df_temp = df_temp.reset_index().set_index('TIME')
+            df_temp = df_temp[parameter_names]
+
+            df_temp = PDresample_by_hour(df_temp, function_dict, function_stats)  # do the magic
+            df_temp['instrument_index'] = np.repeat(file_index, len(df_temp)).astype('int32')
+            df_data = pd.concat([df_data, df_temp.reset_index()], ignore_index=True, sort=False)
+
+        file_index += 1
+
+    df_metadata.index.rename('INSTRUMENT', inplace=True)
+    df_data.index.rename('OBSERVATION', inplace=True)
+    ## rename index to TIME
+    df_data.rename(columns={'index': 'TIME'}, inplace=True)
+
+
+    nc_metadata = xr.Dataset({'LONGITUDE': (['INSTRUMENT'], df_metadata['LONGITUDE'].astype('float32')),
+                              'LATITUDE': (['INSTRUMENT'], df_metadata['LATITUDE'].astype('float32')),
+                              'NOMINAL_DEPTH': (['INSTRUMENT'], df_metadata['NOMINAL_DEPTH'].astype('float32')),
+                              'instrument_id': (['INSTRUMENT'], df_metadata['instrument_id'].astype('|S256')),
+                              'source_file': (['INSTRUMENT'], df_metadata['source_file'].astype('|S256'))})
+
+    nc_data = xr.Dataset.from_dataframe(df_data)
+    nc_aggregated = xr.merge([nc_metadata, nc_data])
+
+
+    ## add global attributes
+    add_attribute = {'rejected_files': "\n".join(list(bad_files))}
+    nc_aggregated.attrs = set_globalattr(nc_aggregated, globalattr_file, site_code, add_attribute, parameter_names)
+
+
+    ## add variable attributes
+    variablenames_others = ['TIME', 'LONGITUDE', 'LATITUDE', 'NOMINAL_DEPTH',
+                            'instrument_index', 'instrument_id', 'source_file']
+    add_variable_attribute = {'PRES_REL': {'applied_offset_by_instrument': applied_offset}}
+    parameter_names_all = list(set(parameter_names_all))
+    variable_attributes = set_variableattr(parameter_names_all + variablenames_others, variable_attribute_dictionary,
+                                           add_variable_attribute)
```

Comment:
I get a `KeyError` in `set_variableattr()` in a test case where none of my input files have a `PRES_REL` variable. Looking at it more closely, I realised you don't need this function at all! Just read the attributes from the JSON template file straight into `variable_attributes`, and update it with the additional PRES_REL attribute.

Reply:
You're right! changed.